crates/matrix-sdk-ui/src/notification_client.rs

```diff
@@ -543,23 +536,43 @@ impl NotificationItem {
 
         let sender = match room.state() {
             RoomState::Invited => room.invite_details().await?.inviter,
-            _ => {
-                if sync_members {
-                    room.get_member(event.sender()).await?
-                } else {
-                    room.get_member_no_sync(event.sender()).await?
-                }
-            }
+            _ => room.get_member_no_sync(event.sender()).await?,
         };
 
-        let (sender_display_name, sender_avatar_url) = match sender {
+        let (mut sender_display_name, mut sender_avatar_url) = match &sender {
             Some(sender) => (
                 sender.display_name().map(|s| s.to_owned()),
                 sender.avatar_url().map(|s| s.to_string()),
             ),
             None => (None, None),
         };
 
+        // XXX(bnjbvr): (remove before merging) Very manual processing here, but it's
+        // unclear if we should hold it in the state store.
```

Comment:
Must be addressed :-). Just a reminder.

Reply:
That's actually an open question: is there a better way to automate this kind of processing? I've seen that `process_sync` could be able to do it, but it expects a full sync response, which we don't have after receiving a `/context` query. We could *forge* one response, but it might be incomplete and we'd need to be careful to not break anything elsewhere. WDYT?

Reply:
Closing that one; @poljar was of the opinion that's it's fine to do the manual processing of state events here :+1:

Reply:
OK :-)